addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/ContentSecurityPolicyMissingScannerUnitTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.httpclient.URI;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Plugin;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class ContentSecurityPolicyMissingScannerUnitTest
+        extends PassiveScannerTest<ContentSecurityPolicyMissingScanner> {
+    private static final String URI = "https://www.example.com";
+    private static final String HEADER_HTML = "Content-Type: text/html";
+    private static final String HEADER_TEXT = "Content-Type: text/plain";
+    private static final String HEADER_CSP = "Content-Security-Policy: x";
+    private static final String HEADER_X_CSP = "X-Content-Security-Policy: x";
+    private static final String HEADER_WEBKIT_CSP = "X-WebKit-CSP: x";
+    private static final String HEADER_REPORT_ONLY = "Content-Security-Policy-Report-Only: x";
+
+    @Override
+    protected ContentSecurityPolicyMissingScanner createScanner() {
+        return new ContentSecurityPolicyMissingScanner();
+    }
+
+    @Test
+    public void givenMissingCspHeaderThenAlertRaised() throws Exception {
+        // Given
+        HttpMessage msg = createHttpMessageWithHeaders(new String[] {HEADER_HTML});
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertContentSecurityPolicyAlertRaised();
+    }
+
+    @Test
+    public void givenOnlyCspHeaderAtMediumAlertThresholdThenNoAlertRaised() throws Exception {
```

Comment:
Was the `Only` added on purpose? `CspHeader` seems enough.

Reply:
It was to indicate that there it was without the browser specific headers.  Can remove if it you think it's more readable without.

Reply:
Since `CspHeader` is singular it already implies it's only one (and CSP).